theseus_gui/src/store/instancesStore.js

```diff
@@ -0,0 +1,337 @@
+          indexSort = 'updated'
+          break
+        default:
+          indexSort = 'relevance'
+      }
+
+      const response = await ofetch(
+        `https://api.modrinth.com/v2/search?query=${this.searchInput || ''}&limit=${
+          this.limit
+        }&offset=${this.offset || 0}${facets || ''}&index=${indexSort}`
+      )
+      this.instances = [...response.hits]
+      this.totalHits = response.total_hits
+      this.offset = response.offset
+      this.pageCount = Math.ceil(this.totalHits / 10)
+    },
+    setSearchInput(newInput) {
+      this.searchInput = newInput
+    },
+    setCurrentPage(newPage) {
+      this.currentPage = newPage
+
+      if (newPage === 1) this.offset = 0
+      else this.offset = this.currentPage * 10 - 10
+    },
+    setFilter(newFilter) {
+      this.filter = newFilter
+    },
+    toggleCategory(cat) {
+      this.categories[cat] = !this.categories[cat]
+    },
+    toggleLoader(loader) {
+      this.loaders[loader] = !this.loaders[loader]
+    },
+    toggleEnv(env) {
+      this.environments[env] = !this.environments[env]
+    },
+    setVersions(versions) {
+      this.activeVersions = versions
+    },
+    setLimit(newLimit) {
+      this.limit = newLimit
+    },
+    resetFilters() {
+      Object.keys(this.categories).forEach((cat) => {
+        this.categories[cat].enabled = false
+      })
+      Object.keys(this.loaders).forEach((loader) => {
+        this.loaders[loader].enabled = false
+      })
+      Object.keys(this.environments).forEach((env) => {
+        this.environments[env] = false
+      })
+      this.activeVersions = []
+    },
+  },
+  getters: {
+    getFilteredInstances: (state) => {
+      const filteredInstances = state.fakeInstances.filter((i) => {
+        // When time comes, do more advanced client-side filtering here if wise
+        const normalizedInstanceName = i.name?.toLowerCase()
+        if (normalizedInstanceName.includes(state.filter.toLowerCase())) return i
+      })
+
+      if (filteredInstances && filteredInstances.length > 0) return filteredInstances
+
+      return state.fakeInstances
+    },
+    getCategoriesByInstanceId: (state) => {
+      return (id) => {
+        let results = []
+        const instance = state.instances?.find((i) => i.project_id === id)
+
+        instance.categories?.forEach((cat) => {
+          // First look for an icon in the categories array
+          let iconObj = generated.categories.find((c) => {
+            if (c.name === cat) return c
+          })
+
+          // If an icon wasn't found in categories, search the loaders array
+          if (!iconObj) {
+            iconObj = generated.loaders.find((l) => {
+              if (l.name === cat) return l
+            })
+          }
+
+          // Push the category for display if an icon was found
+          results.push({ name: cat, icon: iconObj?.icon })
+        })
+
+        return results
+      }
+    },
+    getIconByFilter: (_) => {
```

Comment:
this function seems like it should just be using the `SearchFilter` component. See how it is used here: https://github.com/modrinth/knossos/blob/master/pages/search/%5BsearchProjectType%5D.vue#L41

Reply:
I've slimmed it down and use SearchFilter where possible now. In my current generated file, I don't have icons for some things still, so I've kept getIconByFilter as a fallback until that's resolved.